### ISSUE TITLE
restorePoller forgotten-export client context to the top level 

### ISSUE THREAD

**Issue by qiaozha** [Azure/autorest.typescript#2605]
<img width="1107" alt="image" src="https://github.com/Azure/autorest.typescript/assets/55688292/f2c9a3a7-e290-4aa7-95d4-f5b415d60945">

we have used the client context type in restorePoller, but it's not being exported. see https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/review/arm-networkanalytics.api.md

**Comment by MaryGao** [comment]
This is because we didn't export `NetworkAnalyticsContext` which is in REST-layer client to top-level. Let's talk about the options:

- Option 1: Ignore this warning

- Option 2: Remove this `NetworkAnalyticsContext` from restorePoller

- Option 3: Replace `NetworkAnalyticsContext` with rest-client `Client`

**Comment by joheredi** [comment]
There is Option 4:

- Re-export the context from Rest but not directly. At the api layer we could do this in the index.ts file
```ts
import {NetworkAnalyticsContext } from "../rest/index.js";
export {NetworkAnalyticsContext };
```

That way it is not exported directly from the rest layer, but still exported. The end result is the same, but we keep things organized. I don't think we should do **Option 1** b/c it could be confusing trying to figure out where to get it if someone needs it. **Option 2** causes friction with casting which I'd like to avoid and **Option 3** is the general purpose client which is not very helpful as there is no guidance for the user on how to construct it.

I think Option 4 should be good but open to suggestions

**Comment by MaryGao** [comment]
Offline confirmed we agreed with option 2 in preview stage and need to revisit when GA.
